apps/mobile/src/modules/documents-actions/components/document-action-sheet.tsx

```diff
@@ -0,0 +1,319 @@
+
+                {/* Document details */}
+                <View style={styles.detailsContainer}>
+                  <View style={styles.detailRow}>
+                    <MaterialCommunityIcons
+                      name="file"
+                      size={14}
+                      color={themeColors.mutedForeground}
+                      style={styles.detailIcon}
+                    />
+                    <Text style={styles.detailText}>{formatFileSize(document.originalSize)}</Text>
+                  </View>
+                  <View style={styles.detailRow}>
+                    <MaterialCommunityIcons
+                      name="calendar"
+                      size={14}
+                      color={themeColors.mutedForeground}
+                      style={styles.detailIcon}
+                    />
+                    <Text style={styles.detailText}>{formatDate(document.createdAt)}</Text>
+                  </View>
+                  <View style={styles.detailRow}>
+                    <MaterialCommunityIcons
+                      name="file-document-outline"
+                      size={14}
+                      color={themeColors.mutedForeground}
+                      style={styles.detailIcon}
+                    />
+                    <Text style={styles.detailText} numberOfLines={1}>
+                      {document.mimeType.split('/')[1] || document.mimeType}
+                    </Text>
+                  </View>
+                </View>
+              </View>
+
+              {/* Action buttons */}
+              <View style={styles.actions}>
+                {isViewable && (
+                  <TouchableOpacity
+                    style={styles.actionButton}
+                    onPress={() => {
+                      onClose();
+                      onView();
+                    }}
+                    activeOpacity={0.7}
+                  >
+                    <View style={[styles.actionIcon, styles.viewIcon]}>
+                      <MaterialCommunityIcons
+                        name="eye"
+                        size={20}
+                        color={themeColors.primary}
+                      />
+                    </View>
+                    <Text style={styles.actionText}>View</Text>
+                  </TouchableOpacity>
+                )}
+
+                <TouchableOpacity
+                  style={styles.actionButton}
+                  onPress={() => {
+                    onClose();
+                    handleDownloadAndShare();
+                  }}
+                  activeOpacity={0.7}
+                >
+                  <View style={[styles.actionIcon, styles.downloadIcon]}>
+                    <MaterialCommunityIcons
+                      name="download"
+                      size={20}
+                      color={themeColors.primary}
+                    />
+                  </View>
+                  <Text style={styles.actionText}>Download & Share</Text>
```

Comment:
[wording]
Here `Download ` is more a matter of technical implementation here, people may just want to share 

```suggestion
                  <Text style={styles.actionText}>Share</Text>
```